### PR TITLE
fix page permissions for video

### PR DESCRIPTION
**Pull request by mattcasey**
Added support for 'post' type pages when checking permissions on video. There's a lot of changes because I fixed an import to the middleware folder.

### CODE REVIEW

File: components/common/CharmEditor/components/floatingMenu/Icon.tsx [hunk 1]
```diff
@@ -10,7 +10,9 @@ export type MenuButtonProps = {
   onMouseDown?: React.MouseEventHandler;
 };
 
-const StyledMenuButton = styled(ListItemButton)<{ active: boolean }>`
+const StyledMenuButton = styled(ListItemButton, { shouldForwardProp: (prop) => prop !== 'active' })<{
```
Comment (mattcasey) [inline components/common/CharmEditor/components/floatingMenu/Icon.tsx:13]: this just fixes a random warning in the dev console

File: components/common/CharmEditor/components/inlinePalette/editorItems/media.tsx [hunk 2]
```diff
@@ -45,7 +45,7 @@ export function items(): PaletteItemTypeNoGroup[] {
       uid: 'video',
       title: 'Video',
       icon: <VideoLibraryIcon sx={{ fontSize: iconSize }} />,
-      description: 'Insert a video block in the line below',
+      description: 'Insert a video block',
```
Comment (mattcasey) [inline components/common/CharmEditor/components/inlinePalette/editorItems/media.tsx:48]: make the description fit on one row

File: lib/middleware/index.ts [hunk 3]
```diff
@@ -6,5 +6,4 @@ export * from './requireKeys';
 export * from './requireApiKey';
 export * from './requireSuperApiKey';
 export * from './requireSudoApiKey';
-export * from '../users/hasAccessToSpace';
```
Comment (mattcasey) [inline lib/middleware/index.ts:9]: We're linking to a file outside of the index.ts's folder. Maybe this changed when someone moved the file from middleware to lib/users and VS Code auto-fixed imports?

File: lib/mux/__tests__/permissions.spec.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import { createUserFromWallet } from 'lib/users/createUser';
+import { generateUserAndSpaceWithApiToken } from 'testing/setupDatabase';
```
Comment (mattcasey) [inline lib/mux/__tests__/permissions.spec.ts:2]: tests here

**Review by msadura — APPROVED** [review]

**Review by motechFR — APPROVED** [review]

**Pull request merged**